### PR TITLE
[ticket/12505] Remove outdated media handling in attachments

### PR DESCRIPTION
**Pull request by n-aleha**
Remove code for handling media in RealPlayer, QuickTime, and Windows Media
Player.

[PHPBB3-12505](https://tracker.phpbb.com/browse/PHPBB3-12505)
~~https://github.com/phpbb/phpbb/pull/2462~~

### CODE REVIEW

**Comment by marc1706** [comment]
@Nicofuma shouldn't we add a migration to update the existing board with the database changes?

File: phpBB/includes/constants.php [hunk 1]
```diff
@@ -171,11 +171,11 @@
 // Categories - Attachments
 define('ATTACHMENT_CATEGORY_NONE', 0);
 define('ATTACHMENT_CATEGORY_IMAGE', 1); // Inline Images
-define('ATTACHMENT_CATEGORY_WM', 2); // Windows Media Files - Streaming
-define('ATTACHMENT_CATEGORY_RM', 3); // Real Media Files - Streaming
+define('ATTACHMENT_CATEGORY_WM', 2); // Windows Media Files - Streaming - @deprecated 3.1
```
Comment (Nicofuma) [inline phpBB/includes/constants.php]: @deprecated 3.2 now :)

**Comment by Nicofuma** [comment]
Updating the database could be a good idea but I don't remember if we store this information somewhere in the existing attachments. Do we?

**Comment by n-aleha** [comment]
The main obstacle of this PR is how to migrate existing settings with respect to an admin's choices.
There is some talk about it in the previous PR #2462 and a migration attempt https://gist.github.com/n-aleha/8258c160cfd51f185304

Currently, we keep the attachment Special Categories as constants but the actual Extension Groups are in the db. We also keep in the attachment table, only the file extension but I don't think we need to touch the attachments table.

The most compromising solution will be to keep the Extension Groups for: Quicktime Media, Real Media and Windows Media and change their special category to None. 
If we adopt that, I should change the commit https://github.com/n-aleha/phpbb/commit/993cf9c42cb412b08d3483629050585f44c65750 and not move all those extensions to Downloadable Files.

I should probably squash the last commit https://github.com/n-aleha/phpbb/commit/a292f625d01092fec8a0d33eaf66d71680e7f0bd too.

**Comment by Nicofuma** [comment]
Moving everything to download-able sounds like a good idea to me.
But....

- If a user hasn't enabled an outdated extension group, we could simply delete it
- If a board admin has deleted the donwloadable files group... I don't know

Or we can do what you describe for the existing installation and remove everything for the new one.

@marc1706 thoughts?

**Comment by marc1706** [comment]
Yeah, just change the special category to none and move those types to downloadable files.

Btw, you currently get this in the acp:

```
[phpBB Debug] PHP Notice: in file [ROOT]/includes/acp/acp_attachments.php on line 895: Undefined offset: 6
[phpBB Debug] PHP Notice: in file [ROOT]/includes/acp/acp_attachments.php on line 895: Undefined offset: 3
[phpBB Debug] PHP Notice: in file [ROOT]/includes/acp/acp_attachments.php on line 895: Undefined offset: 2
```

**Comment by marc1706** [comment]
@n-aleha How are things looking? Care to fix the mentioned issues?

**Comment by n-aleha** [comment]
Let me summarize this.
For new installations we keep none of the extensions.
See here, https://github.com/n-aleha/phpbb/commit/993cf9c42cb412b08d3483629050585f44c65750 for the ones that we keep.

For existing installations, if the outdated extension groups exists, change all file extensions group (even the ones added by a user) to category none and move them to downloadable files.

And if the admin has deleted the downloadable files group?

**Comment by marc1706** [comment]
In that case you should set the extension group ID to 0 which means that no group is assigned.

**Comment by n-aleha** [comment]
I've updated this. It should be ready for a 1st review.

File: phpBB/phpbb/db/migration/data/v320/remove_outdated_media.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+			array('custom', array(array($this, 'change_extension_group'))),
+		);
+	}
+
+	public function change_extension_group()
+	{
+		// select group ids of outdated media
+		$sql = 'SELECT group_id
+			FROM ' . EXTENSION_GROUPS_TABLE . '
+			WHERE ' . $this->db->sql_in_set('cat_id', $cat_id);
```
Comment (marc1706) [inline phpBB/phpbb/db/migration/data/v320/remove_outdated_media.php]: Needs to be $this->cat_id

**Comment by marc1706** [comment]
You can test a migration by checking out master, installing the board, merging your PR, and afterwards running install/database_update.php

**Comment by n-aleha** [comment]
Updated.The migration file seems to work.
Thanks for the migration tip @marc1706 
With version upgrade, do we need to clear the extensions cache as we do when we delete an extension group from acp? See, https://github.com/phpbb/phpbb/blob/master/phpBB/includes/acp/acp_attachments.php#L658

**Comment by marc1706** [comment]
Yes, please do that in the migration, too.

**Comment by Elsensee** [comment]
IMHO, no. Cache is purged after an update. This includes the extensions cache too, of course.

**Comment by Nicofuma** [comment]
you need to rebase

About the cache, it is purged when using database_update.php and when using db:migrate

**Comment by n-aleha** [comment]
Rebased.
The conflict was in the following commit because of event addition: https://github.com/n-aleha/phpbb/commit/436b020f9e93eb19f52fa8ce65f6b596d4d7f20d

**Comment by Nicofuma** [comment]
Looks good to me now, @marc1706 ?

**Pull request merged**